HW2/P4/driver.py

```diff
@@ -13,17 +13,45 @@
 
 import filtering
 from timer import Timer
-import threading
+from threading import Semaphore, Thread
+
+def parallel_median_3x3(tmpA, tmpB, iters, line, num_threads, semaphores):
+    for i in range(iters):
+        # Make sure to loop around for the top thread semaphore
+        top = semaphores[(line + 1) % num_threads][0]
+        top.acquire()
+
+        bottom = semaphores[line - 1][1]
+        bottom.acquire()
+
+        filtering.median_3x3(tmpA, tmpB, line, num_threads)
+        tmpA, tmpB = tmpB, tmpA
+
+        semaphores[line][0].release()
+        semaphores[line][1].release()
```

Comment:
Oh cool! Interesting to see how you use semaphores.  I've only seen Events until now.